EIPS/eip-6774.md

```diff
@@ -0,0 +1,262 @@
+- `safeTransferFrom(address,address,uint256)` (`0x42842e0e`)
+- `transferFrom(address,address,uint256)` (`0x23b872dd`)
+
+## Rationale
+
+This EIP has been designed to leverage existing features of the [ERC-721](./eip-721.md) such as `approve/allowance` and `Transfer` events and leave the full control on the implementation.
+
+### Permissionless barter
+
+Tokens created compliant with this EIP MUST design authorizations between tokens on both `barter` and `transferFor` functions to keep the non-sellability of tokens. We explored cases of permissionless or one-way permissionless barter, in both cases it opened up the possibility of a bypassed one-way transfer.
+
+#### Fully permissionless
+
+Allowing any barter type between NSTs make no such sense to prevent the non-sellability of the token. Any smart contract could bypass this kind of transfer by creating a valueless NST to barter with others. In this case, replacing transfer functions would be useless.
+
+The value equivalence is so maintained by tokens creators as a whitelist, keeping these authorisations at the contract level is important and would help create a network of barterable NSTs.
+
+Based on value equivalence we leverage the game theory to enforce the non-sellability instead of stronger restrictions or centralization which would lead to higher code complexity and poor interoperability. Indeed, the strong counterpart at the transfer made NSTs highly illiquid to be subject to a sale on a marketplace or by OTC agreement.
+
+#### One way permissionless
+
+##### On `transferFor`
+
+Leaving this method permissionless would lead to the possibility to self transfer the token to any other address, without a barter.
+
+##### On `barter`
+
+Using `barter` without restriction opens up the risk of unwanted barters between non-equivalent NSTs. Thus any valueless NST could be used to perform a one-way transfer.
+
+### Multi NST contract barter
+
+Implementing a many-to-many NST contract is quite impossible as the barter occurs at the contract level, therefore implement such feature require to call and write on each NST contract. The gas usage for such a feature would be too expensive.
+
+### Register of allowed NST
+
+Creating a universal register for barter allowances between NST contracts (like the [ERC-1820](./eip-1820.md)) would be hard to maintain and easy to bypass.
+
+### Third party protocol
+
+Implementing a service for barter NFT is already existent (see Sudoswap OTC). Such a service still enable one-way transfer through ERC-721 properties and thus not prevent the sellability of the token.
+
+## Backwards Compatibility
+
+Other than the disabled methods specified in [Contract interface](#contract-interface), this EIP is backward compatible with [ERC-721](./eip-721.md).
+
+## Test Cases
+
+The reference implementation includes unit tests written using Foundry.
+
+## Reference Implementation
+
+[Implementations examples](../assets/eip-6774/) have been designed for this proposal.
```

Comment:
Unfortunately our renderer doesn't like linking to directories, so you'll need to link to a specific file. One option that works fairly well for big reference implementations is creating a `README.md` and linking to that.